src/App.tsx

```diff
@@ -15,36 +16,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  alphabet = 'alphabet',
+  length = 'length',
+  default = '',
+}
+
+function prepareGoods(
+  goods: string[],
+  sortedField: string,
+  isReversed: boolean,
+) {
+  const copyGoods = [...goods];
+
+  if (sortedField !== SortType.default) {
+    copyGoods.sort((elementA, elementB) => {
+      switch (sortedField) {
+        case SortType.alphabet:
+          return elementA.localeCompare(elementB);
+
+        case SortType.length:
+          return elementA.length - elementB.length;
+
+        case SortType.default:
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    copyGoods.reverse();
+  }
+
+  return copyGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortedField, setSortField] = useState(SortType.default);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = prepareGoods(goodsFromServer, sortedField, isReversed);
+
+  function handleReset() {
```

Comment:
It is better to use the arrow function to avoid issues with scope as App component is an arrow function